### DISCUSSION TITLE
A note on Kanban embeds

### DISCUSSION THREAD

**Discussion by mgmeyers** [zsviczian/obsidian-excalidraw-plugin#141] [General]
@zsviczian I finally had some time to look into why embedded drawings don't work reliably in kanban boards. The crux of the issue is that, for performance reasons, the Kanban plugin pre-renders markdown, then inserts it into the dom by cloning the rendered markdown node: https://github.com/mgmeyers/obsidian-kanban/blob/bffe5c088a0c4d91bb847427ed7f9a5658998fd4/src/components/MarkdownRenderer.tsx#L89

The problem is that there's a race condition between this and your markdownPostProcessor here: 

https://github.com/zsviczian/obsidian-excalidraw-plugin/blob/784a7cb6bfea10dd50e5b21530a465f446617020/src/main.ts#L225-L247

The `async` function you have in `createDiv` can finish execution after your `markdownPostProcessor` has run, which means that the kanban plugin will clone the node before the image is appended. If you moved `const img = await getIMG(attr);` outside of this function, and made the function synchronous, it should get drawings to load properly in kanban boards.

**Comment by zsviczian** [comment]
Awesome! This resolved the issue.
I'll be publishing an update later today.
